BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
+			
+        var data = client.GetStringAsync(MainCurrenciesUrl).Result;
```

Comment:
It can be wise to add a try-catch block around the interaction with the external API. We then have more control of the outcome, perhaps we would like to translate certain exceptions to our clients rather than serving a 500, or if the client seem unreliable we may want to exercise a retry policy. Have you given any thoughts into this?

Reply:
You are using .Result here rather than an await for the asynchronous method, which stalls the thread until completion instead of the enclosing async method which would let other asynchronous methods propagate in the meantime. It's not necessarily wrong in this context but most often it is better to make use of the await keyword and make this method asynchronous to better cope with load.